pkg/dataobj/index/builder.go

```diff
@@ -251,19 +251,23 @@ func (p *Builder) run(ctx context.Context) error {
 	level.Info(p.logger).Log("msg", "started index builder service")
 	for {
 		fetches := p.client.PollRecords(ctx, -1)
-		if fetches.IsClientClosed() || ctx.Err() != nil {
-			return ctx.Err()
-		}
-		if errs := fetches.Errors(); len(errs) > 0 {
```

Comment:
This would discard records for all fetches in the case where at least one (but not all) fetches returned an error.